airbyte-ci/connectors/pipelines/pipelines/commands/groups/connectors.py

```diff
@@ -361,6 +362,75 @@ def publish(
     return all(context.state is ContextState.SUCCESSFUL for context in publish_connector_contexts)
 
 
+@connectors.command(cls=DaggerPipelineCommand, help="Publish selected connectors to PyPI.")
+@click.option(
+    "--pypi-username",
+    help="Your username to connect to PyPI.",
+    type=click.STRING,
+    required=True,
+    envvar="PYPI_USERNAME",
+)
+@click.option(
+    "--pypi-password",
+    help="Your password to connect to PyPI.",
+    type=click.STRING,
+    required=True,
+    envvar="PYPI_PASSWORD",
+)
+@click.option(
+    "--pypi-repository",
+    help="The PyPI repository to publish to (pypi, test-pypi).",
+    type=click.Choice(["pypi", "testpypi"]),
+    default="pypi",
+)
+@click.pass_context
+def publish_pypi(
+    ctx: click.Context,
+    pypi_username: str,
+    pypi_password: str,
+    pypi_repository: str,
+):
+    selected_connectors_and_files = ctx.obj["selected_connectors_and_files"]
+    selected_connectors_names = ctx.obj["selected_connectors_names"]
+
+    main_logger.info(f"Will publish the following connectors to PyPI: {', '.join(selected_connectors_names)}")
+
+    connectors_contexts = [
+        PublishPyPIContext(
+            connector=connector,
+            is_local=ctx.obj["is_local"],
+            git_branch=ctx.obj["git_branch"],
+            git_revision=ctx.obj["git_revision"],
+            modified_files=modified_files,
+            ci_report_bucket=ctx.obj["ci_report_bucket_name"],
+            report_output_prefix=ctx.obj["report_output_prefix"],
+            gha_workflow_run_url=ctx.obj.get("gha_workflow_run_url"),
+            dagger_logs_url=ctx.obj.get("dagger_logs_url"),
+            pipeline_start_timestamp=ctx.obj.get("pipeline_start_timestamp"),
+            ci_context=ctx.obj.get("ci_context"),
+            ci_gcs_credentials=ctx.obj["ci_gcs_credentials"],
+            pypi_username=pypi_username,
+            pypi_password=pypi_password,
+            pypi_repository=pypi_repository,
+        )
+        for connector, modified_files in selected_connectors_and_files.items()
+    ]
```

Comment:
```suggestion
    connectors_contexts = [
        PublishPyPIContext(
            connector=connector,
            is_local=ctx.obj["is_local"],
            git_branch=ctx.obj["git_branch"],
            git_revision=ctx.obj["git_revision"],
            ci_report_bucket=ctx.obj["ci_report_bucket_name"],
            report_output_prefix=ctx.obj["report_output_prefix"],
            gha_workflow_run_url=ctx.obj.get("gha_workflow_run_url"),
            dagger_logs_url=ctx.obj.get("dagger_logs_url"),
            pipeline_start_timestamp=ctx.obj.get("pipeline_start_timestamp"),
            ci_context=ctx.obj.get("ci_context"),
            ci_gcs_credentials=ctx.obj["ci_gcs_credentials"],
            pypi_username=pypi_username,
            pypi_password=pypi_password,
            pypi_repository=pypi_repository,
        )
        for connector in ctx.obj["selected_connectors_with_modified_files"]
    ]
```

Following the changes of https://github.com/airbytehq/airbyte/pull/28855